### PR TITLE
add readme to .github

### LINKED ISSUE #5483: [CT-848] Add .github README
Add a `README` to `.github` with standards for our workflows/actions

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #5483 

### Description

Add standards and tips for working with github actions in dbt-core

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Review by ChenyuLInx — COMMENTED** [review]
This looks great!! Excited to have more readmes

File: .github/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    run: aws s3 cp ${{ env.s3_bucket }} . --recursive 
+  ```
+
+### Testing
+
+- Depending on what your action does, you may be able to use [act](https://github.com/nektos/act) to test the action locally.  Some features of GitHub actions do not work with act, among those are reusable workflows.  If you can't use act, you'll have to push your changes up before being able to test.  This can be slow.
+
+- View _all_ action output from the [actions](https://github.com/dbt-labs/dbt-core/actions) tab.  Workflow results last 1 year.
+
+- View the action output for your PR in the **Checks** tab of the PR.  This only shows the most recent run.
```
Comment (dbeatty10) [inline .github/README.md:143]: Would it be useful to have consistent casing and bold formatting for the labs of the "Actions" and "Checks" tabs?
```suggestion
- View _all_ action output from the [**Actions**](https://github.com/dbt-labs/dbt-core/actions) tab.  Workflow results last 1 year.

- View the action output for your PR in the **Checks** tab of the PR.  This only shows the most recent run.
```

File: .github/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+- [GitHub Actions guides](https://docs.github.com/en/actions/guides)
+
+## dbt Labs Action Repo
+[dbt-labs/actions](https://github.com/dbt-labs/actions/) is a central repository of actions and workflows dbt uses across repositories.
+
+## General Standards
+
+### Permissions
+- default to least permissions
+- there are a lot of permission.  [Read up on them](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) if you're unsure what to use.
```
Comment (dbeatty10) [inline .github/README.md:16]: Is "permission" intended to be singular here? Or plural?
```suggestion
- there are a lot of permissions.  [Read up on them](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) if you're unsure what to use.
```

File: .github/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        echo The changelog_path: ${{ inputs.changelog_path }}
+        echo The build_script_path: ${{ inputs.build_script_path }}
+        echo The s3_bucket_name: ${{ inputs.s3_bucket_name }}
+        echo The package_test_command: ${{ inputs.package_test_command }}
+  ```
+
+## Tips
+
+### Context
+- The GitHub cli is available in the default runners
```
Comment (dbeatty10) [inline .github/README.md:110]: Suggestion for consistent uppercasing of "CLI" and addition of external URL:
```suggestion
- The [GitHub CLI](https://cli.github.com/) is available in the default runners
```

File: .github/README.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
+## What are GitHub Actions?
+
+GitHub Actions are used for many different purposes.  We use them to run tests in CI, validate PRs are in an expected state, and automate processes.
+
+- [Overview of GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions)
+- [What's a workflow?](https://docs.github.com/en/actions/using-workflows/about-workflows)
+- [GitHub Actions guides](https://docs.github.com/en/actions/guides)
+
+## dbt Labs Action Repo
```
Comment (leahwicz) [inline .github/README.md:9]: This makes it sound like all our Actions live in this repo. Maybe this section could be- Where do our Actions live? Then you can point here but also point locally to where you can find specific repo level actions

File: .github/README.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+[dbt-labs/actions](https://github.com/dbt-labs/actions/) is a central repository of actions and workflows dbt uses across repositories.
+
+## General Standards
+
+### Permissions
+- Default to least permissions.
```
Comment (leahwicz) [inline .github/README.md:15]: It might be nice to have a code example of what setting permissions look like. What if I don't set any permissions at all?

Comment (leahwicz) [inline .github/README.md:15]: You also might want to state what the default is which is 
`Workflows have read permissions in the repository for the contents scope only.`

FYI- this is something I'm setting at the repo levels today- I just saw the toggle to do this so we might have to fix some actions that don't abide by this currently

File: .github/README.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+- There are a lot of permissions available.  [Read up on them](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) if you're unsure what to use.
+
+### using on_merge?
+- TODO: what was the security hole related to this?
```
Comment (leahwicz) [inline .github/README.md:19]: I'm not what you are referring to here? I can message you to get more context here

File: .github/README.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    
+### Secrets
+- When to use a [Personal Access Token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) vs the [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) generated for the action?
+
+    If you expect the resulting commit to add a changelog should retrigger workflows, you will need to use a Personal Access Token for the bot to commit the file. When using the GITHUB_TOKEN, the resulting commit will not trigger another GitHub Actions Workflow run. This is due to limitations set by GitHub. See [the docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow) for a more detailed explanation.
```
Comment (leahwicz) [inline .github/README.md:24]: I would state that you are using the `GITHUB_TOKEN` by default and that in most cases that is all you need. For chaining workflow runs together (i.e. workflow A triggers workflow B), those are special cases where GitHub doesn't want to cause an infinite loop of workflows triggering and that is when you need to explicitly use a PAT

**Review by leahwicz — APPROVED** [review]
Thanks for writing this up Emily! This is a good summary of best practices and knowledge to share

**Pull request merged**